test/gateway/gateway_base/prove_gateway.js

```diff
@@ -0,0 +1,97 @@
+    let stateRoot = "0x70b4172eb30c495bf20b5b12224cd2380fccdd7ffa2292416b9dbdfc8511585d";
+    let storageRoot = '0x8c0ee0843488170879578464b1cadcdb7377efa787372405ff373e4cec6a56db';
+
+    beforeEach(async function () {
+
+      let core = await Core.new(1, 2, 0, stateRoot, accounts[1])
+        , organisation = accounts[2]
+        , messageBus = accounts[1]
+        , bounty = new BN(100);
+
+      gatewayBaseInstance = await GatewayBase.new(
+        core.address,
+        messageBus,
+        bounty,
+        organisation
+      );
+
+    });
+
+    it('should be able to prove with correct request', async function () {
+
+      let response = await gatewayBaseInstance.proveGateway(1, rlpEncodedAccount, rlpParentNodes);
+
+      let expectedEvent = {
+        GatewayProven: {
+          _gateway: 0,
+          _blockHeight: new BN(1),
+          _storageRoot: storageRoot,
+          _wasAlreadyProved: false
+        }
+      };
+
+      assert.equal(
+        response.receipt.status,
+        1,
+        "Receipt status is unsuccessful"
+      );
+
+      let eventData = response.logs;
+      await Utils.validateEvents(eventData, expectedEvent);
+    });
+
+    it('should be not prove  with if already proven', async function () {
+
+      await gatewayBaseInstance.proveGateway(2, rlpEncodedAccount,
+        rlpParentNodes);
+
+      let response = await gatewayBaseInstance.proveGateway(2, rlpEncodedAccount, rlpParentNodes);
+
+      let expectedEvent = {
+        GatewayProven: {
+          _gateway: 0,
+          _blockHeight: new BN(2),
+          _storageRoot: storageRoot,
+          _wasAlreadyProved: true
+        }
+      };
+
+      assert.equal(
+        response.receipt.status,
+        1,
+        "Receipt status is unsuccessful"
+      );
+
+      let eventData = response.logs;
+
+      await Utils.validateEvents(eventData, expectedEvent);
```

Comment:
how does this assert that it is not proven again when it actually checks for the event (compare title)? 😕